modules/UI/util/MessageHandler.js

```diff
@@ -113,6 +115,74 @@ function isDontShowAgainEnabled(options) {
     return typeof options === 'object';
 }
 
+/**
+ * By default how long a notification should display before being automatically
+ * dismissed.
+ *
+ * @private
+ * @type {number}
+ */
+const NOTIFICATION_DISMISS_TIMEOUT = 2500;
+
+/**
+ * An array of names of participants that have joined the conference. The array
+ * is replaced with an empty array as notifications are displayed.
+ *
+ * @private
+ * @type {string[]}
+ */
+let joinedParticipantsNames = [];
+
+/**
+ * A throttled internal function that takes the internal list of participant
+ * names, {@code joinedParticipantsNames}, and triggers the display of a
+ * notification informing of their joining.
+ *
+ * @private
+ * @type {Function}
+ */
+const throttledNotifyParticipantConnected = throttle(() => {
```

Comment:
I'm always a little bit concerned when we add new stuff to the non-react files. This sounds like it belongs to the participants feature in react. Do you see any problem with making it an action there and only dispatching it here?

Reply:
I've moved the logic into participant action like you wanted. Please confirm this truly is what you want.